docs/reference/api/platform/Logger.md

```diff
@@ -0,0 +1,164 @@
+
+Usage:
+```C
+Funcitons returns NULL and does not terminate the application. If error is categorized as fatal, use `MBED_CRIT` instead.
+void *operator new(std::size_t count) {
+    void *buffer = malloc(count);
+    if (NULL == buffer) {
+        MBED_ERR("OS", "Operator new out of memory");
+    }
+    return buffer;
+}
+
+Fatal error, will terminate the application if memory is not available.
+void *operator new(std::size_t count) {
+    void *buffer = malloc(count);
+    if (NULL == buffer) {
+        MBED_CRIT("OS", "Operator new out of memory");
+    }
+    return buffer;
+}
+```
+
+#### LOG_LEVEL_WARN
+You should use warning level in OS, HAL, library, application, etc to log all the warning messages.
+
+Usage:
+```C
+void dev_init() {
+    ...
+    if (DEV_ID != id) {
+        MBED_WARN ("Driver", "Incorrect Device ID");
+    }
+    ...
+}
+```
+
+#### LOG_LEVEL_DEBUG
+You should use debug level in user application for adding debug level messages. You should not use debug level in OS, HAL and libraries instead trace level should be used. 
+
+Usage:
+```C
+void main() {
+    ...
+    int err = dev_init();
+    if (SUCCESS == err) {
+        MBED_DBG("Main", "Init successful");
+    }
+    ...
+}
+```
+You can alse use `MBED_DBG_IF` function for debug level logs. It is similar to `MBED_DBG` except that it takes an additional argument for condition, the message is logged only if condition evaluates to true.
+
+Usage:
+```C
+void main() {
+    ...
+    int err = dev_init();
+    MBED_DBG_IF("Main", (SUCCESS == err), "Init successful");
+    ...
+}
+```
+
+#### LOG_LEVEL_INFO
+You can use info level messages to log any additional information. 
+
+Usage:
+```C
+void main() {
+    ...
+    case STATE_DISCONNECTED:
+        MBED_INFO("Main", "Disconnected");
+        break;
+    ...
+}
+```
+You can alse use `MBED_INFO_IF` function for info level logs. It is similar to `MBED_INFO` except that it takes an additional argument for condition, the message is logged only if condition evaluates to true.
+
+Usage:
+```C
+void main() {
+    ...
+    case STATE_DISCONNECTED:
+        MBED_INFO_IF("Main", (true == wasRequested), "Disconnect successful");
+        MBED_INFO_IF("Main", (false == wasRequested), "Disconnect event occurred");
+        break;
+    ...
+}
+```
+
+#### LOG_LEVEL_TRACE
+You should use trace level in HAL, OS and libraries for adding debug logs. Trace level messsages are by default disabled, you can set `MBED_CONF_MAX_LOG_LEVEL` as `LOG_LEVEL_TRACE` to enable all trace logs.
+
+Usage:
+```C
+int dev_init() {
+    ...
+    MBED_TRACE("Driver", "Init successful");
+    return SUCCESS;
+}
+```
+You can alse use `MBED_TRACE_IF` function for trace level logs. It is similar to `MBED_TRACE` except that it takes an additional argument for condition, the message is logged only if condition evaluates to true.
+
+Usage:
+```C
+#define ENABLE_DRIVER_MSG   0
```

Comment:
Is it possible to enable all levels of messages from a specific driver( may be a  single .c file ) but disable everything else?

Reply:
Please read the note section below, if all drivers implement it right way then yes.

Reply:
You cannot do that for error / critical and warn level though

Reply:
Ok, Thanks for clarifying.